python/semantic_kernel/planners/plan.py

```diff
@@ -200,9 +200,12 @@ def metadata(self) -> KernelFunctionMetadata:
     def set_available_functions(self, plan: "Plan", kernel: "Kernel", arguments: "KernelArguments") -> "Plan":
         if len(plan.steps) == 0:
             try:
-                pluginFunction = kernel.plugins[plan.plugin_name][plan.name]
-                plan.set_function(pluginFunction)
-            except Exception:
+                plugin_function = kernel.plugins[plan.plugin_name][plan.name]
```

Comment:
let's use `kernel.get_function` for this and then catch _KernelFunctionNotFoundError_ and _KernelPluginNotFoundError_ only

Reply:
Thanks for reviewing! I've pushed the changes.